source/_addonStore/dataManager.py

```diff
@@ -89,15 +100,21 @@ def __init__(self):
 		self._compatibleAddonCache = self._getCachedAddonData(self._cacheCompatibleFile)
 		self._installedAddonsCache = _InstalledAddonsCache()
 		# Fetch available add-ons cache early
-		threading.Thread(
+		self._getAddonsThread = threading.Thread(
 			target=self.getLatestCompatibleAddons,
 			name="initialiseAvailableAddons",
-		).start()
+			daemon=True,
+		)
+		self._getAddonsThread.start()
+
+	def terminate(self):
+		if self._getAddonsThread.is_alive():

```

Comment:
As suggested by @michaelDCurran it would be nice to know that a given thread didn't finish in a reasonable time and preferably know what it was doing when NVDA exits. I'd suggest either adding additional logging here so that it is clear that we are killing it. as an alternative we can just rely on the fact that it is a daemon and will be killed eventually and add a logging to the end of `core.main` listing active threads and what they're doing using `getFormattedStacksForAllThreads` as an example of what can be logged.